dist/index.js

```diff
@@ -55343,24 +55346,34 @@ async function getRegistrationToken() {
   }
 }
 
-async function removeRunner() {
-  const runner = await getRunner(config.input.label);
+// remove all runners with the input label
+async function removeRunners() {
   const octokit = github.getOctokit(config.input.githubToken);
 
-  // skip the runner removal process if the runner is not found
-  if (!runner) {
-    core.info(`GitHub self-hosted runner with label ${config.input.label} is not found, so the removal is skipped`);
+  // find all runners with our label
+  const allRunners = await octokit.paginate('GET /repos/{owner}/{repo}/actions/runners', config.githubContext);
+  const runners = _.filter(allRunners, { labels: [{ name: config.input.label }] });
+
+  // skip the runner removal process if the runners are not found
+  if (runners.length < 1) {
+    core.info(`GitHub self-hosted runner(s) with label ${config.input.label} not found, so the removal is skipped`);
     return;
   }
 
-  try {
-    await octokit.request('DELETE /repos/{owner}/{repo}/actions/runners/{runner_id}', _.merge(config.githubContext, { runner_id: runner.id }));
-    core.info(`GitHub self-hosted runner ${runner.name} is removed`);
-    return;
-  } catch (error) {
-    core.error('GitHub self-hosted runner removal error');
-    throw error;
+  let failed = 0;
+  for (let runner of runners) {
+    try {
+      await octokit.request('DELETE /repos/{owner}/{repo}/actions/runners/{runner_id}', _.merge(config.githubContext, { runner_id: runner.id }));
+      core.info(`GitHub self-hosted runner ${runner.name} (${runner.id}) is removed`);
+    } catch (error) {
+      core.error(`GitHub self-hosted runner ${runner.name} (${runner.id}) removal error: ${error}`);
+      failed++;
+    }
+  }
+  if (failed > 0) {
+    throw `Failed to shutdown ${failed} of ${runners.length} GitHub self-hosted runners`;
```

Comment:
The consequence is that that the action logs the exception as an error message and fails the step e.g.

```
AWS EC2 instance i-023050a07a0de1c2c is terminated
::error::GitHub self-hosted runner i-023050a07a0de1c2c (365) removal error: HttpError: Not Found
::error::Failed to shutdown 1 of 1 GitHub self-hosted runners
::error::
```

That is the same as the action is currently, just extended to multiple runner. The current code is:

```javascript
  try {
    await octokit.request('DELETE /repos/{owner}/{repo}/actions/runners/{runner_id}', _.merge(config.githubContext, { runner_id: runner.id }));
    core.info(`GitHub self-hosted runner ${runner.name} is removed`);
    return;
  } catch (error) {
    core.error('GitHub self-hosted runner removal error');
    throw error;
```

Also, it wont block the deploy, because the deploy step is not dependant on the `stop-runner`, it depends on the run-specs step. The shutdown just runs in parallel to the deploy steps.